subsys/bluetooth/audio/cap_initiator.c

```diff
@@ -1692,34 +1748,259 @@ int bt_cap_initiator_unicast_audio_stop(const struct bt_cap_unicast_audio_stop_p
 
 	for (size_t i = 0U; i < param->count; i++) {
 		struct bt_cap_stream *cap_stream = param->streams[i];
+		struct bt_bap_stream *bap_stream = &cap_stream->bap_stream;
+
+		/* Ensure that ops are registered before any procedures are started */
+		bt_cap_stream_ops_register_bap(cap_stream);
 
 		active_proc->proc_param.initiator[i].stream = cap_stream;
-	}
+		active_proc->proc_param.initiator[i].stop.release = param->release;
 
-	bt_cap_common_start_proc(BT_CAP_COMMON_PROC_TYPE_STOP, param->count);
+		if (!can_disable && can_disable_stream(bap_stream)) {
```

Comment:
Somehow missing the link to my original comment. Anyway good point that this is in a loop.
It still could be rewritten as

'''
 can_disable = can_disable || can_disable_stream(bap_stream);
'''

which still would do the short circuiting, but I am not sure if that's more readable than what you have now

Reply:
Hmm, I think the current version is more explicit. Not sure if there's any performance changes between the two, but I think `can_disable = can_disable || can_disable_stream(bap_stream)` might be worth as it always does 1 assign and 1 compare, whereas the current version always do 1 comparison and optionally an assign.